### PR TITLE
WT-4392 Lint

### PR DESCRIPTION
**Pull request by keithbostic**
Each unrelated change is an individual commit, reviewing the commits is probably the simplest way to review.

### CODE REVIEW

**Comment by agorrod** [comment]
@nehakhatri5 please do a review of this change - it isn't supposed to have any functional differences. I recommend reviewing each commit separately - it will probably make more sense that way.

File: src/include/btmem.h [hunk 1]
```diff
@@ -857,29 +857,6 @@ struct __wt_ref {
 #define	WT_REF_SPLIT	 7		/* Parent page split (WT_REF dead) */
 	volatile uint32_t state;	/* Page state */
 
-#ifdef HAVE_DIAGNOSTIC
```
Comment (agorrod) [inline src/include/btmem.h]: @bvpvamsikrishna you should take a look at the changes in btmem.h from this lint PR.

File: test/format/wts.c [hunk 2]
```diff
@@ -579,15 +579,15 @@ wts_verify(const char *tag)
 void
 wts_stats(void)
 {
+	FILE *fp;
```
[the remark this answers is not in the document]
Comment (keithbostic) [reply test/format/wts.c]: It's in lexicographic sorted order. (I can't defend that, it's just the way it's always been...)

**Comment by nehakhatri5** [comment]
@keithbostic feel free to merge after resolving the merge conflict.

**Comment by agorrod** [comment]
I'm going to go ahead and merge @bvpvamsikrishna I still think it's worth you taking a look at the changes in btmem.h - but I don't see any issues with the code in there.

**Pull request merged**